### DISCUSSION TITLE
Part1 - Ch1 - 8번 terraform 관련 질문

### DISCUSSION THREAD

**Discussion by likespike** [tedilabs/fastcampus-devops#87] [Q&A] [answered]
안녕하세요 Part1 - Ch1 - 8번 terraform 관련 질문있습니다.

설치 중 다음과 같은 오류가 발생합니다
network - terraform apply시 오류가 발생은 했지만 aws에 구성은 된 것 같아
ec2-instance에서 terraform apply해봤는데 여기도 오류가 발생

terraform destroy, init, apply 등 써가며 계속 해보는데 해결이 안됩니다

(왕초보라 상세한 해결 방법 설명 부탁드립니다;;)

감사합니다

<img width="1280" alt="스크린샷 2023-02-21 오후 12 29 18" src="https://user-images.githubusercontent.com/117564349/220242207-086ab2a0-cd9c-498d-a86a-4e360937d7f1.png">

<img width="1280" alt="스크린샷 2023-02-21 오후 12 29 23" src="https://user-images.githubusercontent.com/117564349/220242236-c71ed1b5-8886-46ed-9434-44d25a8fc9d2.png">

<img width="1280" alt="스크린샷 2023-02-21 오후 12 33 42" src="https://user-images.githubusercontent.com/117564349/220242252-c6a180cd-a34f-45bf-b567-c96e6b1ad415.png">

**Comment by posquit0** [comment, marked as answer]
안녕하세요! 제가 요새 개인적인 사정으로 확인이 너무 늦었네요. 죄송합니다.

- 도커 & 쿠버네티스 강좌는 제가 강사진으로 참여한 강의는 아니오나, 패스트캠퍼스 측에서 데브옵스 초격차 패키지의 도커 파트를 일부 기초 섹션에 활용하고 싶다고 요청을 받았습니다. 그러다보니 컨텐츠 자체가 데브옵스 초격차에 엮여있는 것들이 많아 혼란을 초래하고 있는 것 같습니다. 😭
- 사실 도커 컨테이너를 실습하기 위한 리눅스 환경이 필요한 것이라, 해당 실습 코드를 꼭 성공시킬 필요는 없습니다. 가상머신이나 남는 PC 장비에 우분투를 설치하고 docker, docker-compose 등을 설치하여 앞으로 강의를 진행하셔도 괜찮습니다.
- 테라폼 코드를 실제로 실행하여 AWS 상에 인프라 리소스를 구성하시다니 훌륭하십니다. ㅎㅎ 테라폼은 코드의 실제 반영여부를 테라폼 상태(Terraform State)라는 것으로 기억하고, 추후 코드를 변경하여 적용하고자할 경우 테라폼 상태와 코드를 비교하여 변경사항만 적용을 하게 됩니다. 테라폼 상태는 원격 관리와 로컬 관리가 모두 가능한데, 기본값은 로컬 관리(Local States)로 `terraform.tfstate` 파일로 관리합니다. 겪고 있는 오류를 보면 테라폼 코드에서 정의된 리소스가 이미 만들어진 상태에서 또 다시 생성 요청을 하여 발생한 것으로 판단됩니다.
  - 테라폼 코드를 apply 후 제대로 실행을 마무리하지 않았거나하는 상황에서 apply를 실행하면서 발생한 이슈가 아닐까 싶습니다.
  - AWS 계정을 https://github.com/rebuy-de/aws-nuke 와 같은 도구로 완전 초기화, `terraform.tfstate` 파일을 날린 뒤 다시 시도해보시면 잘 동작할 것으로 기대됩니다.
  - AWS 계정 초기화가 번거롭다면 오류 메시지에서 언급하는 이미 존재하는 AWS 리소스를 하나씩 찾아 수동으로 제거해주셔야 합니다.

감사합니다.
